src/Engine/EmitMSIL/Replication.cs

```diff
@@ -2,17 +2,107 @@
 using System.Collections;
 using System.Collections.Generic;
 using System.Linq;
-using System.Text;
-using System.Threading.Tasks;
 using System.Reflection;
 using ProtoCore;
-using ProtoCore.Utils;
 using ProtoCore.Lang.Replication;
+using ProtoCore.Exceptions;
+using ProtoCore.Properties;
+using ProtoCore.DSASM;
+using ProtoCore.Utils;
 
 namespace EmitMSIL
 {
     public class Replication
     {
+        internal static List<CLRStackValue> MarshalFunctionArguments(IList args, MSILRuntimeCore runtimeCore)
+        {
+            var marshaller = ProtoFFI.CLRDLLModule.GetMarshaler(runtimeCore);
+            List<CLRStackValue> stackValues = new List<CLRStackValue>();
+            foreach (var arg in args)
+            {
+                var protoType = ProtoFFI.CLRObjectMarshaler.GetProtoCoreType(arg.GetType());
+                CLRStackValue dsObj = marshaller.Marshal(arg, protoType, runtimeCore);
+                stackValues.Add(dsObj);
+            }
+            return stackValues;
+        }
+
+        internal static List<object> UnmrshalFunctionArguments(List<CLRFunctionEndPoint.ParamInfo> formalParams, List<CLRStackValue> args, MSILRuntimeCore runtimeCore)
```

Comment:
`UnmrshalFunctionArguments` is copied from the CLRFFIFunctionPointer.Execute function (the marshaling/conversion parts)